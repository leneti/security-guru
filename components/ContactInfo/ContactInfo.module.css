.wrapper {
  display: flex;
  align-items: center;
  color: white;
}

.icon {
  background-color: transparent;
}

.title {
  color: var(--mantine-color-dark-0);
  font-weight: bold;
}

.description {
  color: var(--mantine-color-dark-1);
}

.icon {
  color: var(--mantine-primary-color);
}
