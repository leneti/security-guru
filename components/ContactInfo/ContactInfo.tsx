import {
  createStyles,
  Text,
  Box,
  useMantineTheme,
  SimpleGrid,
} from "@mantine/core";
import { contactData } from "@site/constants";

const useStyles = createStyles((theme) => ({
  wrapper: {
    display: "flex",
    alignItems: "center",
    color: theme.white,
  },

  icon: {
    marginRight: theme.spacing.md,
    backgroundColor: "transparent",
  },

  title: {
    color: theme.colors.dark[0],
    fontWeight: "bold",
  },

  description: {
    color: theme.colors.dark[2],
  },
}));

export default function ContactInfo() {
  const { classes } = useStyles();
  const theme = useMantineTheme();

  const items = contactData.map(({ icon: Icon, title, description }) => (
    <div key={title} className={classes.wrapper}>
      <Box mr="md">
        <Icon size={24} color={theme.fn.primaryColor()} />
      </Box>

      <div>
        <Text size="xs" className={classes.title}>
          {title}
        </Text>
        <Text className={classes.description}>{description}</Text>
      </div>
    </div>
  ));

  return (
    <SimpleGrid
      cols={1}
      spacing="md"
      breakpoints={[
        { maxWidth: "sm", cols: 2 },
        { maxWidth: "xs", cols: 1 },
      ]}
    >
      {items}
    </SimpleGrid>
  );
}
