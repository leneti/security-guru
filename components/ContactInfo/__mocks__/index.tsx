export function ContactInfo() {
  return <div>Mock ContactInfo</div>;
}
