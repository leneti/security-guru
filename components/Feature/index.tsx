export { default as Feature } from "./Feature";
export * from "./types";
