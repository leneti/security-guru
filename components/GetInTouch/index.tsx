export { default, type ContactForm } from "./GetInTouch";
