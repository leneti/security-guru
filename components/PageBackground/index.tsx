export { default, type PageBGProps } from "./PageBackground";
