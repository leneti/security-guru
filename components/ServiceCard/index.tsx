export { default as ServiceCard } from "./ServiceCard";
export * from "./types";
