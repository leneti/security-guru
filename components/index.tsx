export { default as ContactInfo } from "./ContactInfo";
export { default as Footer } from "./Footer";
export { default as GetInTouch, type ContactForm } from "./GetInTouch";
export { default as Header } from "./Header";
export { default as Hero } from "./Hero";
export { default as Logo } from "./Logo";
export { default as PageTitle } from "./PageTitle";
export { default as PageBackground, type PageBGProps } from "./PageBackground";
export { default as SlideDownTitle } from "./SlideDownTitle";
