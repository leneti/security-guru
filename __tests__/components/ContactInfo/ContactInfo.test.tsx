import { render } from "@testing-library/react";
import { ContactInfo } from "@site/components/ContactInfo";

jest.mock(
  "@tabler/icons",
  () =>
    new Proxy(
      {
        __esModule: true,
      },
      {
        get: (_: any, prop: string) => () => <div>mock_{prop}</div>,
      }
    )
);

describe("ContactInfo", () => {
  it("renders", () => {
    const { container } = render(<ContactInfo />);
    expect(container).toMatchSnapshot();
  });
});
