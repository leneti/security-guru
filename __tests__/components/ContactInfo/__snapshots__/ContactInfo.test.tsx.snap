// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo renders 1`] = `
<div>
  <div
    class="mantine-SimpleGrid-root mantine-185uj33"
  >
    <div
      class="mantine-12v0vsc"
    >
      <div
        class="mantine-zidq9q"
      >
        <div>
          mock_
          IconAt
        </div>
      </div>
      <div>
        <div
          class="mantine-Text-root mantine-1dnllud"
        >
          El. paštas
        </div>
        <div
          class="mantine-Text-root mantine-1ylbvz9"
        >
          info@securityguru.lt
        </div>
      </div>
    </div>
    <div
      class="mantine-12v0vsc"
    >
      <div
        class="mantine-zidq9q"
      >
        <div>
          mock_
          IconPhone
        </div>
      </div>
      <div>
        <div
          class="mantine-Text-root mantine-1dnllud"
        >
          Tel. Nr.
        </div>
        <div
          class="mantine-Text-root mantine-1ylbvz9"
        >
          +37060334255, +37060283726
        </div>
      </div>
    </div>
    <div
      class="mantine-12v0vsc"
    >
      <div
        class="mantine-zidq9q"
      >
        <div>
          mock_
          IconMapPin
        </div>
      </div>
      <div>
        <div
          class="mantine-Text-root mantine-1dnllud"
        >
          Dirbame
        </div>
        <div
          class="mantine-Text-root mantine-1ylbvz9"
        >
          Vilniuje ir Vilniaus apskrityje
        </div>
      </div>
    </div>
    <div
      class="mantine-12v0vsc"
    >
      <div
        class="mantine-zidq9q"
      >
        <div>
          mock_
          IconSun
        </div>
      </div>
      <div>
        <div
          class="mantine-Text-root mantine-1dnllud"
        >
          Darbo laikas
        </div>
        <div
          class="mantine-Text-root mantine-1ylbvz9"
        >
          I-VII 9:00-21:00
        </div>
      </div>
    </div>
    <div
      class="mantine-12v0vsc"
    >
      <div
        class="mantine-zidq9q"
      >
        <div>
          mock_
          IconBuildingEstate
        </div>
      </div>
      <div>
        <div
          class="mantine-Text-root mantine-1dnllud"
        >
          Įm. kodas
        </div>
        <div
          class="mantine-Text-root mantine-1ylbvz9"
        >
          306109454
        </div>
      </div>
    </div>
  </div>
</div>
`;
